app/src/main/java/com/android/sample/model/map/MapViewModel.kt

```diff
@@ -0,0 +1,95 @@
+package com.android.sample.model.map
+
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import com.android.sample.ui.map.CameraAction
+import com.google.android.gms.maps.model.LatLng
+import kotlinx.coroutines.launch
+
+val defaultLocation = LatLng(46.51915277948766, 6.566736625776037)
+
+open class MapViewModel(val mapLocationRepository: LocationRepository) : ViewModel() {
```

Comment:
would call this "LocationViewModel" or "MapLocationViewModel" to make it more precise